src/server.go

```diff
@@ -76,9 +77,38 @@ func (a *App) QueryHandler(key []byte, w http.ResponseWriter, r *http.Request) {
   }
 }
 
+func (a *App) GetAuthorization(w http.ResponseWriter, r *http.Request) (string, bool) {
+  var authstring string
+  if a.basicauth == true {
+    authtoken := r.Header.Get("Authorization")
+    if authtoken != "" {
+      authstringb, _ := base64.StdEncoding.DecodeString(strings.Split(authtoken, "Basic ")[1])
+      authstring = string(authstringb)
+      authstring = authstring+"@"
+      remote := fmt.Sprintf("http://%s%s", authstring, a.volumes[rand.Intn(len(a.volumes))])
+      resp, err := http.Get(remote)
+      if err != nil {
+        w.WriteHeader(500)
+        return authstring, false
+      }
+      if resp.StatusCode != 200 {
+        w.WriteHeader(resp.StatusCode)
+        return authstring, false
+      }
+    }
+  }
+  return authstring, true
+}
+
 func (a *App) ServeHTTP(w http.ResponseWriter, r *http.Request) {
   key := []byte(r.URL.Path)
 
+  // check if user is authorized
```

Comment:
I'd put the if a.basicauth in here. It's more readable that way, and GetAuthorization shouldn't have to depend on the app class. Follow principle of least scope.